src/git.rs

```diff
@@ -1,23 +1,59 @@
-use git2::Repository;
+use git2::{Repository, Commit, Oid};
 use std::fs::File;
 use std::io::prelude::*;
+use crate::fingerprints::Fingerprint;
 use std::error::Error;
+use std::process::*;
+use crate::error::CapnError;
+use std::str;
+use std::collections::HashSet;
+
 
 use crate::config::*;
+use log::*;
 
 pub trait Git: Sized {
     fn new() -> Result<Self, Box<dyn Error>>;
     fn read_file(&self, path: &str) -> Result<String, Box<dyn Error>>;
     fn write_git_file(&self, path: &str, file_mode: u32, contents: &str) -> Result<(), Box<dyn Error>>;
-    
     fn current_branch(&self) -> Result<String, Box<dyn Error>>;
+    fn log(&self) -> Result<(), Box<dyn Error>>;
+    fn find_commits(&self,from_id: Oid,to_id: Oid) -> Result<Vec<Commit<'_>>, Box<dyn Error>>;
+    fn find_unpushed_commits(&self, new_commit_id: Oid) -> Result<Vec<Commit<'_>>, Box<dyn Error>>;
+    fn find_commit(&self,commit_id: Oid) -> Result<Commit<'_>, Box<dyn Error>>;
+    fn find_commit_fingerprints(&self, team_fingerprint_file: &str, commits: &Vec<Commit<'_>>) -> Result<HashSet<String>, Box<dyn Error>>;
+    fn find_common_ancestor(&self, commit1_id: Oid, commit2_id: Oid) -> Result<Oid, Box<dyn Error>>;
+    fn pushed(&self,commit_id: Oid) -> Result<bool, Box<dyn Error>>;
+    fn single_commit(commit: &Commit<'_>) -> Result<bool, Box<dyn Error>>;
```

Comment:
The naming here is a bit confusing.  I'm not sure what a "single_commit" is.
Perhaps you should rename it to "not_merge_commit"?